.github/workflows/evaluation-test-reusable.yaml

```diff
@@ -0,0 +1,193 @@
+name: "Evaluation tests"
+run-name: "Evaluation tests"
+
+on:
+  workflow_call:
+
+# global env variables.
+env:
+  DOCKER_TIMEOUT: 30
+  K3D_VERSION: "v5.7.2" # Required K3d version.
+  REPOSITORY_FULL_NAME: "${{ github.repository }}" # <owner>/<repository-name>.
+  GITHUB_EVENT_ACTION: "${{ github.event.action }}"
+  IMAGE_NAME: "europe-docker.pkg.dev/kyma-project/dev/kyma-companion:PR-${{ github.event.number }}"
+
+jobs:
+  wait-for-build:
+    name: Wait for image build job
+    runs-on: ubuntu-latest
+    if: contains(github.event.pull_request.labels.*.name, 'evaluation requested')
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          ref: main
+
+      - name: Setup python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.10"
+          cache: "pip"
+
+      - name: Install requirements
+        run: |
+          pip install -r ./scripts/python/wait-for-commit-check/requirements.txt
+
+      - name: wait for build
+        env:
+          GITHUB_TOKEN: ${{ github.token }}
+          # Note: 'github.event.pull_request.head.sha' is not same as 'github.sha' on pull requests.
+          GIT_REF: ${{ github.event.pull_request.head.sha }}
+          REPOSITORY_FULL_NAME: "${{ github.repository }}"
+          # The re-useable image-builder workflow from neighbors appends the "Build image" suffix to the check run name.
+          GIT_CHECK_RUN_NAME: "build / Build image"
+          INTERVAL: 60
+          TIMEOUT: 900
+        run: |
+          python ./scripts/python/wait-for-commit-check/run.py
+
+  evaluation-tests:
+    name: Run evaluation tests
+    needs: wait-for-build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
```

Comment:
The ref and repository should also be input parameters, so we can change the source of evaluation tests.